api/resources_portal/test/views/test_material.py

```diff
@@ -215,6 +224,17 @@ def test_delete_request_deletes_a_material(self):
         self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
         self.assertEqual(Material.objects.filter(id=material_id).count(), 0)
 
+        self.assertEqual(
+            len(Notification.objects.filter(notification_type="MATERIAL_DELETED")),
+            self.organization.members.count(),
+        )
+
+    def test_delete_material_with_request_fails(self):
+        MaterialRequestFactory(material=self.material)
+        self.client.force_authenticate(user=self.user)
+        response = self.client.delete(self.url)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```

Comment:
I would consider making this error a 403? 400 gives the impress that the actual syntax of the request is wrong.

Reply:
Hmm, not sure I agree. 

> 400 Bad Request response status code indicates that the server cannot or will not process the request due to something that is perceived to be a client error (e.g., malformed request syntax, invalid request message framing, or deceptive request routing

- https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400

I would consider this to fall under the category of `invalid request message framing`. Basically the request is invalid because you cannot delete a material with requests. I think a 403 might imply that someone was allowed to? (I don't think a 403 would be wrong here necessarily, but IMO 400 is better.)